tests/models/encoder_decoder/vision_language/test_mllama.py

```diff
@@ -7,22 +7,31 @@
 from vllm.multimodal.utils import rescale_image_size
 from vllm.sequence import SampleLogprobs
 
-from ....conftest import (IMAGE_ASSETS, HfRunner, PromptImageInput, VllmRunner,
-                          _ImageAssets)
+from ....conftest import HfRunner, PromptImageInput, VllmRunner, _ImageAssets
 from ....utils import large_gpu_test
 from ...utils import check_logprobs_close
 
-_LIMIT_IMAGE_PER_PROMPT = 1
-
-HF_IMAGE_PROMPTS = IMAGE_ASSETS.prompts({
-    "stop_sign":
-    "<|image|><|begin_of_text|>The meaning of the image is",
-    "cherry_blossom":
-    "<|image|><|begin_of_text|>The city is",
-})
-
-text_only_prompts = [
-    "The color of the sky is blue but sometimes it can also be",
+_LIMIT_IMAGE_PER_PROMPT = 2
+
+# ("prompt", [image indices])
+# image-0: stop sign, image-1: cherry blossom
+# For each entry, we will generate a batch of
+# samples with different image sizes.
+PROMPTS = [
+    # Single leading image.
+    ("<|image|><|begin_of_text|>The meaning of the image is", [0]),
+    ("<|image|><|begin_of_text|>The city is", [1]),
+    # Single interleaved image.
+    ("<|begin_of_text|>The meaning of the image<|image|> is", [0]),
+    # Multi leading images.
+    ("<|image|><|image|><|begin_of_text|>Between the first and second image, "
+     "which is stop sign and which is cherry blossom?", [0, 1]),
+    # Multi interleaved images.
+    ("<|begin_of_text|>Between the first image<|image|> and second "
+     "image<|image|>, which is stop sign and which is cherry blossom?", [0,
+                                                                         1]),
+    # Text only.
+    ("The color of the sky is blue but sometimes it can also be", []),
```

Comment:
Can you write the unit tests with similar structure to other models, e.g., `test_llava_onevision.py` to make the code easy to be maintained? And I think it would be more clear if you can split different features into different tests.

Reply:
done